### PR TITLE
Initial pass at adding some tables, associations

### PR DESCRIPTION
**Pull request by Timo614**
I've started adding some tables and associations here. I refactored a bit of the db logic to break out the models into their own files to make it a bit easier to parse and encapsulated their db related logic in those models.

Probably should deal with:

- Fixtures for tests / factory if rust has something like that
- A separate test db and db cleaning between runs
- Some unused methods still in this branch from the common source file that was added before (kept them since we're at such an early state we'll be adding functional tests there and didn't seem worth nuking the scaffolding there in light of that)

I'm going to keep working on this branch but wanted to push something out in the meantime.

### CODE REVIEW

**Comment by Timo614** [comment]
I played around a bit trying to figure out a good way to really handle the test db use with cargo running the tests. You can override the test runner in the Cargo toml document as an option but that option didn't seem ideal. Was hoping for a good way to set an environment so that the tests inside of the model logic and the functional and integration all were set as being run in test to rely on that db.

In the branch instead I just added a RUST_ENV env variable which I use via `RUST_ENV=test cargo test` also running migrations via `diesel migration run --database-url=$TEST_DATABASE_URL`. http://docs.diesel.rs/diesel/connection/trait.Connection.html#method.begin_test_transaction works for keeping the db clean of test data.

Looking into fixtures / better ways of handling test data here and reuse and have a few additional tables I've been adding I'll push up shortly to this branch shortly.

**Comment by CjS77** [comment]
Looking good. I had done very similar migrations before I had to drop everything on Wed.

Some thoughts:
* prefer using uuid instead of INTEGER for ids (for security reasons)
* we can use sequential numbering for the initial migrations to make it clear that this is the initial design (e.g. `00000001-users/up.sql` etc)
* Nit: My gut says that if we're hitting a DB in tests, those tests should be considered integration tests rather than unit tests, and hence live in the `tests` directory.

In summary, keep going. I have some PRs that I'll push soon, which has a little (but not much) overlap; and then we can compare notes.

**Comment by CjS77** [comment]
I've added test DB scaffolding in #7. 
There are a few breaking changes there, so I suggest we manually bring in your changes. 
Overall I think we're rapidly converging to the right basic platform.

**Comment by Timo614** [comment]
Sounds good -- I pushed up some additional changes I had staged up last week as well now.

**Comment by Timo614** [comment]
I'll rebase this branch and address that feedback now.

**Comment by Timo614** [comment]
I have a branch I built off of #7 as a base:
https://github.com/Timo614/bn-db/tree/add-tables

I'll keep rebasing it as #7 changes but it adds the tables there with some adjustments. https://github.com/Timo614/bn-db/commit/1d63bd2541f9f63e209bc7f0ef71ac04c107202d

Going to add tests to it once we figure out how we want to handle the db there. The test db gets nuked each run so wasn't sure if I should remove that if it was stub code or if it was for testing migrations properly are run etc.

**Comment by Timo614** [comment]
Going to close this in favor of my other branch built off of #7 that pulled in some of the changes here / made some additional modifications.

I'll file a PR for that in a few and link back to this one. Getting the API mostly working to use the connection pool and return some basic functionality at the moment so will continue adding to that branch in the meantime.
